aodncore/util/process.py

```diff
@@ -63,7 +63,7 @@ def execute(self):
 
         try:
             proc = subprocess.Popen(self.command, stdin=self.stdin, stdout=self.stdout, stderr=subprocess.STDOUT,
-                                    shell=self.shell, env=self.env)
+                                    shell=self.shell, env=self.env, universal_newlines=True)
```

Comment:
So this will result in `proc.communicate(...)` returning text rather than bytes, below?

Reply:
Yes, that's correct.

https://docs.python.org/3.5/library/subprocess.html#popen-constructor

> If universal_newlines is True, the file objects stdin, stdout and stderr are opened as text streams in universal newlines mode, as described above in Frequently Used Arguments, otherwise they are opened as binary streams.


Note the correctly formatted log message from my pipeline box testing:
```
2020-11-04 14:25:49,532 SYSINFO tasks.SRS_OC_BODBAW[d15a68fd-77f7-4f42-8b49-f8e73bfd7166] get_harvester_runner -> TalendHarvesterRunner()
2020-11-04 14:25:49,533 SYSINFO tasks.SRS_OC_BODBAW[d15a68fd-77f7-4f42-8b49-f8e73bfd7166] harvesting slice size: 2048
2020-11-04 14:25:49,535 SYSINFO tasks.SRS_OC_BODBAW[d15a68fd-77f7-4f42-8b49-f8e73bfd7166] harvester 'srs_oc_bodbaw-srs_oc_bodbaw' matched file: /tmp/SrsOcBodBawHandler887lqtvn/collection/IMOS_SRS-OC-BODBAW_X_20091109T000500Z_SB2009_11-absorption-CDOM_FV02_END-20100722T042400Z_C-20180522T160601Z.nc
2020-11-04 14:25:49,537 INFO tasks.SRS_OC_BODBAW[d15a68fd-77f7-4f42-8b49-f8e73bfd7166] running additions for harvester 'srs_oc_bodbaw-srs_oc_bodbaw'
2020-11-04 14:25:49,540 SYSINFO tasks.SRS_OC_BODBAW[d15a68fd-77f7-4f42-8b49-f8e73bfd7166] executing /usr/local/talend/jobs/srs_oc_bodbaw-srs_oc_bodbaw/java/SRS_OC_BODBAW_harvester/SRS_OC_BODBAW_harvester_run.sh --context_param paramFile="/usr/local/talend/jobs/srs_oc_bodbaw-srs_oc_bodbaw/etc/srs_oc_bodbaw-srs_oc_bodbaw.conf" --context_param base=/tmp/SrsOcBodBawHandler887lqtvn/temp/talend_baseq__k31tn --context_param fileList=/tmp/SrsOcBodBawHandler887lqtvn/temp/talend_baseq__k31tn/file_list3o9odixu.txt --context_param logDir=/mnt/ebs/log/pipeline/process
2020-11-04 14:25:49,541 INFO tasks.SRS_OC_BODBAW[d15a68fd-77f7-4f42-8b49-f8e73bfd7166] --- START TALEND OUTPUT ---
Warning: Parameter "base" has not been set by Implicit_Context_Context
Warning: Parameter "paramFile" has not been set by Implicit_Context_Context
Warning: Parameter "logDir" has not been set by Implicit_Context_Context
Warning: Parameter "fileList" has not been set by Implicit_Context_Context
Start of try block
tempFile_cid /tmp/talend-temp-3270719972619828259/changelog.xml
***********************
conn host:hostname, port:5432, schema:srs_oc_bodbaw, name:harvest?loginTimeout=1000&ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory, user:srs_oc_bodbaw
url_cid:jdbc:postgresql://hostname:5432/harvest?loginTimeout=1000&ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory
schema_cid:srs_oc_bodbaw
userName_cid:srs_oc_bodbaw
***** completed finally block ***********
In finally block
Exception in component iPostgresqlDbUpdate_1 (HarvestMetadataFromFileSystem)
org.postgresql.util.PSQLException: FATAL: password authentication failed for user "srs_oc_bodbaw"
        at org.postgresql.Driver$ConnectThread.getResult(Driver.java:348)
        at org.postgresql.Driver.connect(Driver.java:271)
        at java.sql.DriverManager.getConnection(DriverManager.java:664)
        at java.sql.DriverManager.getConnection(DriverManager.java:247)
        at srs_oc_bodbaw.harvestmetadatafromfilesystem_0_1.HarvestMetadataFromFileSystem.iPostgresqlDbUpdate_1Process(HarvestMetadataFromFileSystem.java:1438)
        at srs_oc_bodbaw.harvestmetadatafromfilesystem_0_1.HarvestMetadataFromFileSystem.runJobInTOS(HarvestMetadataFromFileSystem.java:4205)
        at srs_oc_bodbaw.harvestmetadatafromfilesystem_0_1.HarvestMetadataFromFileSystem.runJob(HarvestMetadataFromFileSystem.java:3989)
        at srs_oc_bodbaw.srs_oc_bodbaw_harvester_0_1.SRS_OC_BODBAW_harvester.tRunJob_1Process(SRS_OC_BODBAW_harvester.java:1497)
        at srs_oc_bodbaw.srs_oc_bodbaw_harvester_0_1.SRS_OC_BODBAW_harvester$2.run(SRS_OC_BODBAW_harvester.java:5948)
Exception in component tRunJob_1 (SRS_OC_BODBAW_harvester)
java.lang.RuntimeException: Child job running failed.
org.postgresql.util.PSQLException: FATAL: password authentication failed for user "srs_oc_bodbaw"
        at srs_oc_bodbaw.srs_oc_bodbaw_harvester_0_1.SRS_OC_BODBAW_harvester.tRunJob_1Process(SRS_OC_BODBAW_harvester.java:1514)
        at srs_oc_bodbaw.srs_oc_bodbaw_harvester_0_1.SRS_OC_BODBAW_harvester$2.run(SRS_OC_BODBAW_harvester.java:5948)

--- END TALEND OUTPUT ---
```

It's a little fiddly to test, but the new unit test *does* confirm that the output is the expected text, since byte strings do not compare as equal to unicode strings:

```python
b'string' == 'string'
False
```

So given this, and the expected result in the "real" log, I'd say it's pretty safe.

Reply:
Cool :+1: 
Thanks!